### PR TITLE
Ormlite Example

### PR DESCRIPTION
**Pull request by C3H6O3**
An example of @OrmLiteDao

### CODE REVIEW

**Comment by WonderCsabo** [comment]
Thanks for your contribution!

@yDelouis @dodgex do you think we need a whole example project for this? Injecting DAOs is pretty straightforward, and this example has more code showing OrmLite then just the DAO injection.

**Comment by dodgex** [comment]
I do Not think that such an example project is needed. The wiki should contain the Info needed to get the injection done and general usage should be teached by ormlite Website.

**Comment by WonderCsabo** [comment]
@C3H6O3 did you read the `@OrmLiteDao` [wiki page](https://github.com/excilys/androidannotations/wiki/Ormlite)? Do you think it needs more information as you wish to add this example project?

**Comment by yDelouis** [comment]
I don't think either that an example project is needed.
We just add the injection which is quite simple and for anything else, this is the documentation of OrmLite that should be improved.

**Comment by WonderCsabo** [comment]
@yDelouis do you think sg is missing from the doc?

**Comment by yDelouis** [comment]
I don't. What I meant is that, if one does, he should ask on OrmLite channels.

**Comment by WonderCsabo** [comment]
OK. 

@C3H6O3 Thanks for your contribution, but it seems we reached a consensus to not merge this PR. I will still add some comments on your code, so you will have some feedback if you want to post it somewhere.

File: examples/OrmliteExample/.idea/compiler.xml [hunk 1]
```diff
@@ -0,0 +1,23 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (WonderCsabo) [inline examples/OrmliteExample/.idea/compiler.xml]: The whole .idea folder and iml files are not needed. They are redundant, and generated from the gradle project descriptor which contains all information.

File: examples/OrmliteExample/app/build.gradle [hunk 2]
```diff
@@ -0,0 +1,37 @@
+apply plugin: 'com.android.application'
+apply plugin: 'android-apt'
+def AAVersion = '3.2'
```
Comment (WonderCsabo) [inline examples/OrmliteExample/app/build.gradle]: We released AA 3.3 now. :smirk:

File: examples/OrmliteExample/app/build.gradle [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+    buildTypes {
+        release {
+            minifyEnabled false
+            proguardFiles getDefaultProguardFile('proguard-android.txt'), 'proguard-rules.pro'
+        }
+    }
+}
+
+dependencies {
+    compile fileTree(dir: 'libs', include: ['*.jar'])
```
Comment (WonderCsabo) [inline examples/OrmliteExample/app/build.gradle]: Not needed, and a bad pattern anyway.

File: examples/OrmliteExample/app/build.gradle [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+    compile 'com.android.support:appcompat-v7:22.1.0'
+    apt "org.androidannotations:androidannotations:$AAVersion"
+    compile "org.androidannotations:androidannotations-api:$AAVersion"
+    compile 'com.j256.ormlite:ormlite-core:4.48'
```
Comment (WonderCsabo) [inline examples/OrmliteExample/app/build.gradle]: You should create a property to make sure the same ormlite version is used, just like for AA.

File: examples/OrmliteExample/app/build.gradle [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+    buildToolsVersion "22.0.1"
+
+    defaultConfig {
+        applicationId "com.example.androidannotations.ormliteexample"
+        minSdkVersion 15
+        targetSdkVersion 22
+        versionCode 1
+        versionName "1.0"
+    }
+    buildTypes {
```
Comment (WonderCsabo) [inline examples/OrmliteExample/app/build.gradle]: The whole block is not needed for the example.

File: examples/OrmliteExample/app/proguard-rules.pro [hunk 6]
```diff
@@ -0,0 +1,17 @@
+# Add project specific ProGuard rules here.
```
Comment (WonderCsabo) [inline examples/OrmliteExample/app/proguard-rules.pro]: Not needed.

File: examples/OrmliteExample/app/src/androidTest/java/com/example/androidannotations/ormliteexample/ApplicationTest.java [hunk 7]
```diff
@@ -0,0 +1,13 @@
+package com.example.androidannotations.ormliteexample;
```
Comment (WonderCsabo) [inline examples/OrmliteExample/app/src/androidTest/java/com/example/androidannotations/ormliteexample/ApplicationTest.java]: The whole `androidTest` should be removed.

File: examples/OrmliteExample/app/src/main/java/com/example/androidannotations/ormliteexample/activity/MainActivity.java [hunk 8]
```diff
@@ -0,0 +1,46 @@
...
+import org.androidannotations.annotations.EActivity;
+import org.androidannotations.annotations.OrmLiteDao;
+import org.androidannotations.annotations.ViewById;
+
+import java.util.Iterator;
+
+@EActivity(R.layout.activity_main)
+public class MainActivity extends Activity {
+
+    @OrmLiteDao(helper = DatabaseHelper.class, model = User.class)
```
Comment (WonderCsabo) [inline examples/OrmliteExample/app/src/main/java/com/example/androidannotations/ormliteexample/activity/MainActivity.java]: Remove the `model` parameter, it is no longer necessary and deprecated.

File: examples/OrmliteExample/app/src/main/java/com/example/androidannotations/ormliteexample/activity/MainActivity.java [hunk 9]
```diff
@@ -0,0 +1,46 @@
...
+
+    @AfterViews
+    void show() {
+        try {
+            User user = userDao.queryForId(1);
+            StringBuilder builder = new StringBuilder();
+            builder.append("User name:" + user.name + "\n");
+            Iterator<Article> iterator = user.articles.iterator();
+            builder.append("Articles:");
+            while (iterator.hasNext()) {
```
Comment (WonderCsabo) [inline examples/OrmliteExample/app/src/main/java/com/example/androidannotations/ormliteexample/activity/MainActivity.java]: Use the for-each loop, it is more readable and more efficient.

File: examples/OrmliteExample/app/src/main/java/com/example/androidannotations/ormliteexample/bean/Article.java [hunk 10]
```diff
@@ -0,0 +1,21 @@
+package com.example.androidannotations.ormliteexample.bean;
+
+import com.j256.ormlite.field.DatabaseField;
+
+/**
```
Comment (WonderCsabo) [inline examples/OrmliteExample/app/src/main/java/com/example/androidannotations/ormliteexample/bean/Article.java]: These default class headers should be removed.

File: examples/OrmliteExample/app/src/main/java/com/example/androidannotations/ormliteexample/bean/Article.java [hunk 11]
```diff
@@ -0,0 +1,21 @@
...
+    public Integer id;
+    @DatabaseField
+    public String title;
+    @DatabaseField(columnName = "user_id", foreign = true)
+    public User user;
+
+    /**
+     * A no-argument constructor is necessary for Ormlite.
+     */
+    public Article() {
```
Comment (WonderCsabo) [inline examples/OrmliteExample/app/src/main/java/com/example/androidannotations/ormliteexample/bean/Article.java]: Nice to tell it to the user. :+1:

File: examples/OrmliteExample/app/src/main/java/com/example/androidannotations/ormliteexample/db/DBUtil.java [hunk 12]
```diff
@@ -0,0 +1,46 @@
...
+ * Created by Administrator on 2015/5/1.
+ */
+@EBean(scope = Scope.Singleton)
+public class DBUtil {
+    @RootContext
+    Context context;
+
+    public void initDatabase() {
+        try {
+            InputStream is = context.getResources().openRawResource(R.raw.test);
```
Comment (WonderCsabo) [inline examples/OrmliteExample/app/src/main/java/com/example/androidannotations/ormliteexample/db/DBUtil.java]: I think it is nicer to just create and persist some example data then to create this class.

File: examples/OrmliteExample/app/src/main/java/com/example/androidannotations/ormliteexample/db/DatabaseHelper.java [hunk 13]
```diff
@@ -0,0 +1,58 @@
...
+     * A single argument(Context) is required.
+     * Because AndroidAnnotations use OpenHelperManager class.
+     */
+    public DatabaseHelper(Context context) {
+        super(context, DATABASE_NAME, null, DATABASE_VERSION);
+        /**
+         * Method1:
+         * To create a new database, you can use getReadableDatabase(); or getWritableDatabase();
+         */
+        //getWritableDatabase();
```
Comment (WonderCsabo) [inline examples/OrmliteExample/app/src/main/java/com/example/androidannotations/ormliteexample/db/DatabaseHelper.java]: This call is not needed actually, the database will be created when needed.

File: examples/OrmliteExample/app/src/main/res/menu/menu_main.xml [hunk 14]
```diff
@@ -0,0 +1,6 @@
+<menu xmlns:android="http://schemas.android.com/apk/res/android"
```
Comment (WonderCsabo) [inline examples/OrmliteExample/app/src/main/res/menu/menu_main.xml]: I would remove all things not needed, like all sized pictures, menus, styles, etc. So keep the example as mall as clean, and concentrate on what you demonstrate.

File: examples/OrmliteExample/build.gradle [hunk 15]
```diff
@@ -0,0 +1,15 @@
+buildscript {
+    repositories {
+        mavenCentral()
+    }
+    dependencies {
+        classpath 'com.android.tools.build:gradle:1.1.0'
```
Comment (WonderCsabo) [inline examples/OrmliteExample/build.gradle]: You should use the latest tools and dependencies.

File: examples/OrmliteExample/gradle.properties [hunk 16]
```diff
@@ -0,0 +1,18 @@
+# Project-wide Gradle settings.
```
Comment (WonderCsabo) [inline examples/OrmliteExample/gradle.properties]: Should be removed.

File: examples/OrmliteExample/settings.gradle [hunk 17]
```diff
@@ -0,0 +1 @@
+include ':app'
```
Comment (WonderCsabo) [inline examples/OrmliteExample/settings.gradle]: I would not just create one root module, to keep the project small, just like our gradle and RoboGuice examples.

**Comment by C3H6O3** [comment]
Thank you for comments  very much! AndroidAnnotations is awesome.

**Pull request closed without merging**